src/main/java/org/galatea/starter/entrypoint/IexRestController.java

```diff
@@ -48,4 +49,18 @@ public List<IexLastTradedPrice> getLastTradedPrice(
     return iexService.getLastTradedPriceForSymbols(symbols);
   }
 
+  /**
+   * Get historical pricing data for the given symbol on the given date
+   *
+   * @param symbol the symbol to retrieve data about
+   * @param date the date which should be queried
+   * @return historical pricing data for the given symbol on the given date
+   */
+  @GetMapping(value = "${mvc.iex.getHistoricalPricePath}", produces = {
+          MediaType.APPLICATION_JSON_VALUE})
+  public List<IexHistoricalPrice> getHistoricalPrice(
+          @RequestParam(value = "symbol") final List<String> symbol,
+          @RequestParam(value = "date") final List<String> date) {
```

Comment:
There are certain expectations on the values we eventually pass to the IEX API, e.g. the provided date should be valid. Try adding some validations so that if there is invalid data sent in, we can fail the request early.

Reply:
When replacing line 63 with:
>@RequestParam(value = "date") @DateTimeFormat(pattern = "yyyyMMdd") final List<LocalDate> dates)
Spring performs the validation for me, and will return 500: bad request if an invalid date is entered. Is this best practice or would we be better off accepting the paramter as a string and then performing our own conversion/validations?

Reply:
As a standard way of doing things that's fine.

There could be scenarios in which we may want to perform our own conversions/validations - for example, if we have some complex conversion or validation logic ("accept this pattern in this case, accept this other pattern in this other case"). But generally you'd know if you had to do something like that.